ToDoApp/ToDoApp/ToDoServiceImpl.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  ToDoServiceImpl.swift
+//  ToDoApp
+//
+//  Created by Trong Hieu Vuong on 2022-12-23.
+//
+
+import Foundation
+import FirebaseDatabase
+struct TaskModel: Identifiable, Codable {
+    let id: UUID
+    var title: String
+    var deadline: Date
+    var status: Bool = false
+}
+
+struct ToDoTasks: Codable {
+    var today: [TaskModel]
+    var other: [TaskModel]
+}
+protocol DateChecker {
+    func isDateInToday(_ date: Date) -> Bool
+}
+//  conform Calendar to this protocol
+extension Calendar: DateChecker {}
+
+protocol ToDoService {
+    func createOrUpdateTask(todo: TaskModel) async -> Result<TaskModel, RepositoryError>
```

Comment:
No need to change interfaces of the todo service. It has to be create and update. Because in that case this is dependency on implementation which should not happen according to dependency inversion principle. 
This is firebase feature to create or update objects through one interface. If we would need to replace firebase with another service which wouldn't have that feature our protocol will need to be changed which will break all of the users of that protocol which may have already stable and working code. Which is bad since refactoring may introduce new bugs

Reply:
What I meant earlier is that create task and update task methods had the same implementation except the line on which task object was created in the create method. The idea was too invoke update method with the object created in the create method

Reply:
oh ok, I got it, I've fixed it in the next PR